src/checkers/inference/InferrableAnnotatedTypeFactory.java

```diff
@@ -0,0 +1,22 @@
+package checkers.inference;
+
+import org.checkerframework.framework.type.AnnotatedTypeFactory;
+import org.checkerframework.framework.type.treeannotator.TreeAnnotator;
+
+/**
+ * Interface for all annotatedTypeFactory that wish to be used with
+ * Checker-Framework-Inference
+ */
+
+public interface InferrableAnnotatedTypeFactory {
+
+    /**
+     * return the TreeAnnotator from real type annotatedTypeFactory
+     * 
+     */
+    TreeAnnotator getRealInferenceTreeAnnotator(
```

Comment:
I would remove the "real" from the name. "Real" makes sense from the point-of-view of the InferenceATF. But this is the "real" ATF and it is returning its TreeAnnotator.